flink-table/flink-table-common/src/main/java/org/apache/flink/table/factories/FileSystemFormatFactory.java

```diff
@@ -142,5 +205,17 @@
 					.map(name -> getSchema().getFieldDataType(name).get())
 					.toArray(DataType[]::new);
 		}
+
+		/**
+		 * Get RowType of the table without partition keys.
+		 * @return
+		 */
+		default RowType getNonPartRowType() {
```

Comment:
`getFormatRowType`
And please modify `getFieldNamesWithoutPartKeys ` and `getFieldTypesWithoutPartKeys `